pkg/ccl/backupccl/create_scheduled_backup_test.go

```diff
@@ -659,6 +661,78 @@ func TestSerializesScheduledBackupExecutionArgs(t *testing.T) {
 	}
 }
 
+// TestMixedVersionScheduledBackupDoesNotLockItself is a regression test for a
+// bug where creating a schedule wrote a lock file on the bucket during the "dry
+// run" backup planning. If a different went to execute the backup job, then it
+// would see this lock file and refuse to run.
+func TestMixedVersionScheduledBackupDoesNotLockItself(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	disableUpgradeCh := make(chan struct{})
+	defer close(disableUpgradeCh)
+	var executeSchedule func() error
+	var srv serverutils.TestServerInterface
+	args := base.TestClusterArgs{
+		ServerArgsPerNode: map[int]base.TestServerArgs{
+			0: {Knobs: base.TestingKnobs{
+				Server: &server.TestingKnobs{
+					BinaryVersionOverride:          clusterversion.ByKey(clusterversion.BackupResolutionInJob - 1),
```

Comment:
maybe i'm missing something, but it seems that all three nodes have the same version spec: 
```
BinaryVersionOverride:          clusterversion.ByKey(clusterversion.BackupResolutionInJob - 1),
DisableAutomaticVersionUpgrade: disableUpgradeCh,
```

Reply:
I think that's what we want, we want all the nodes to be performing backup resolution during planning. The node we run `CREATE SCHEDULE` on (node 0) additionally has its job scheduling hijacked so that it doesn't run the actual backup job. When node1/node2 pickup the scheduled job, they should hit the mixed-version BACKUP-LOCK bug.

Reply:
Correct me if I'm wrong, without this diff, we get locked out iff:

1) The node that runs create schedule has upgraded to 22.1, but the cluster version is less than `clusterversion.BackupResolutionInJob - 1`, causing the node to write the lock file during dry run backup planning (very bad). 
2) the node that picks up the scheduled job has also upgraded to 22.1, but the cluster version is less than `clusterversion.BackupResolutionInJob - 1`, causing the node to _plan the back up again_ at a destination that already has a lock file. Because this node plans the backup, it will create a new job ID, so when it sees a different lock file at the target dest, it craps out.

I'm unsure if this actually covers this situation, since each node's _binary_ version is `clusterversion.BackupResolutionInJob - 1`, i.e. each node is on 21.2. But maybe I'm misunderstanding how these test knobs work?

Either way, I think it would be good to spell out exactly when we hit this bug in this commit message and how this test config covers the bug in the test docstring.

Reply:
In unit tests we can't really write true mixed version tests (yet) since all of them are running the binary they are built on. In other words, all the nodes are running 22.1 code in this test, but we override the binary version that the CRDB server thinks it's running so as to force all nodes to see that `clusterversion.BackupResolutionInJob` is not active yet. What you described is the scenario that this test recreates because it will force the node to run the code in `backup_planning.go` that performs all the backup resolution, locking etc because this condition will be false https://github.com/cockroachdb/cockroach/blob/release-22.1/pkg/ccl/backupccl/backup_planning.go#L829.

I'll add some more words to the test comment to explain this.

Reply:
I see, thanks for adding more words!

Reply:
Thanks for the review!